server/src/main/java/org/elasticsearch/search/aggregations/support/ValueType.java

```diff
@@ -80,8 +79,8 @@ public ValuesSourceType getValuesSourceType() {
     }
 
     public boolean isA(ValueType valueType) {
```

Comment:
As you can see from the tests I added later, this is a very generous definition of equivalence.  The modified version keeps the same rules, but makes it a bit more obvious what's going on, IMHO